estate/models/estate_property_offer.py

```diff
@@ -0,0 +1,60 @@
+from dateutil.relativedelta import relativedelta
+
+from odoo import models, fields, api
+from odoo.exceptions import UserError
+
+
+class EstatePropertyOffer(models.Model):
+    _name = "estate.property.offer"
+    _description = "estate property offer"
+    _order = 'price desc'
+
+    deadline = fields.Date(compute="_compute_deadline", inverse="_inverse_validity")
+    price = fields.Float()
+    partner_id = fields.Many2one("res.partner", required=True)
+    property_id = fields.Many2one("estate.property", required=True)
+    property_type_id = fields.Many2one(related="property_id.property_type_id", store=True)
+    validity = fields.Integer("Validity (days)", default=7)
+    status = fields.Selection(
+        selection=[("accepted","Accepted"), ("refused","Refused")], copy=False, readonly=True
+    )
+
+    _sql_constraints =[
+        ('_check_price','CHECK(price > 0)','Offer Price must be positive'),
+    ]
+
+    @api.depends("validity")
+    def _compute_deadline(self):
+        for record in self:
+            record.deadline = fields.Date.today() + relativedelta(days=record.validity) or record.create_date.date() + relativedelta(days=record.validity)
```

Comment:
```suggestion
            record.deadline = record.create_date.date() if record.create_date else fields.Date.today() + relativedelta(days=record.validity)
```
this will always run `fields.Date.today() + relativedelta(days=record.validity) ` we should prioritize create date  suppose
If i have created record yesterday(date=18 for e.g.) and given validity as 2 than the deadline would be tomorrow(date=20).
Now according to your code if i will update validity from 2 to 4 days it should be 22nd as deadline but in your code it will take fields.Date.today() (which is 19th) + 4 days so the deadline would be 23rd.

fields.Date.today() instead check if create_date exists use that else use today because if no create date that means it's a new record.

Reply:
record.deadline = record.create_date.date() + relativedelta(days=record.validity) or fields.Date.today() + relativedelta(days=record.validity)
 
i have changed precedence to the create_date so that this expression evaluated first